tests/unit/test_attribute_mapping_collection.py

```diff
@@ -0,0 +1,127 @@
+from functools import partial
+from unittest import TestCase
+
+import responses
+import simplejson
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+from tamr_unify_client.project.attribute_mapping.collection import (
+    AttributeMappingCollection,
+)
+
+
+class TestAttributeMappingCollection(TestCase):
+    def setUp(self):
+        auth = UsernamePasswordAuth("username", "password")
+        self.unify = Client(auth)
+
+    @responses.activate
+    def test_by_resource_id(self):
+        url = "http://localhost:9100/api/versioned/v1/projects/4/attributeMappings"
+        responses.add(responses.GET, url, json=self.mappings_json)
+        tester = AttributeMappingCollection(self.unify, url)
+        by_resource = tester.by_resource_id("19629-12")
+        return by_resource.unified_attribute_name
+
+    @responses.activate
+    def test_by_relative_id(self):
+        url = "http://localhost:9100/api/versioned/v1/projects/4/attributeMappings"
+        responses.add(responses.GET, url, json=self.mappings_json)
+        tester = AttributeMappingCollection(self.unify, url)
+        by_relative = tester.by_relative_id("projects/4/attributeMappings/19629-12")
+        return by_relative.unified_attribute_name
+
+    @responses.activate
+    def test_create(self):
+        def create_callback(request, snoop):
+            snoop["payload"] = list(request.body)
+            return 200, {}, simplejson.dumps(self._response_json)
+
+        url = "http://localhost:9100/api/versioned/v1/projects/4/attributeMappings"
+        responses.add(responses.GET, url, json=self.mappings_json)
+        responses.add(responses.POST, url, json=self.create_json, status=204)
+        records_url = f"{self.test_by_relative_id}"
+        snoop = {}
```

Comment:
There needs to be a test now that verifies that what's in snoop after the request is made is correct